crates/subspace-networking/src/node.rs

```diff
@@ -272,6 +311,18 @@ impl Node {
         &self,
         key: Multihash,
     ) -> Result<impl Stream<Item = Vec<u8>>, GetValueError> {
+        let kademlia_permit = self
```

Comment:
Can the `shared::Command` be split into something like `shared::{KademliaCommand, GlobalCommand}`, and the corresponding async task in `node_runner::handle_command()` also be split? This way, the semaphores also can be separated, and avoid having to take both of them for kademlia specific tasks

Reply:
Kademlia limits are a subset of global limits, hence I don't think we want to split them. Unless we subtract Kademlia limits from global, which is actually an interesting and probably good idea, but even then I don't see how splitting commands will help as the bottleneck we're considering here is connection level streams in libp2p that are shared between different behavior, including Kademlia.